SimPEG/utils/mat_utils.py

```diff
@@ -411,7 +411,7 @@ def coterminal(theta):
     """
 
     sub = theta[np.abs(theta) >= np.pi]
-    sub = -np.sign(sub) * (2 * np.pi - np.abs(sub))
+    sub = np.sign(sub) * (2 * np.pi - np.abs(sub))
```

Comment:
I don't think this will give you correct values now, simple testing:

```
>>> coterminal(np.pi + np.pi/4)
2.356194490192345  #3pi/4
```
it should give you -3pi/4.

I think the cleanest solution to this is to write the whole function as:

```
def coterminal(theta):
    return (theta + np.pi) % (2 * np.pi) - np.pi
```
Which will give you back a number on the domain of [-\pi, \pi)

Reply:
I agree that using the mod is the cleanest way to solve this.

I would like to see that bugfix in another PR though. I think one of the reasons why this wasn't caught before was due to lack of tests, so besides the fix we should add more tests for it. And also update the docstrings for that function.

Reply:
And why would it be negative you think? That's what it was returning before.
I guess my initial implementation was to return the negative angle of the complement, thinking that if the gradient between two angles is large, we would want to force the model update to reduce both. But now I am not so sure if that logic works.